micrometer-core/src/main/java/io/micrometer/core/instrument/step/StepMeterRegistry.java

```diff
@@ -102,4 +102,15 @@ protected DistributionStatisticConfig defaultHistogramConfig() {
                 .merge(DistributionStatisticConfig.DEFAULT);
     }
 
+    @Override
+    public void close() {
```

Comment:
The default step is `1m` (can be bigger), are you sure everyone wants to `sleep` that much and block `close`?

Also, isn't the original issues for short-lived processes? Isn't blocking their shutdown process counter-intuitive?

Reply:
> The default step is 1m (can be bigger), are you sure everyone wants to sleep that much and block close?

In our use case we don't care how long does it take to shut down the process as long as it publishes the metrics.

> Also, isn't the original issues for short-lived processes? Isn't blocking their shutdown process counter-intuitive?

The processes are probably short-lived not because people want them to be short-lived, but because the work they do is performed fast. At least that's true for our use case.

In our case we have a long running service in a container in Kubernetes, and a cronjob scheduled in Kubernetes as well that uses the same image to do some work. The time the cronjob runs depends on how much work does it have to do, but typically it takes less than a minute. We don't really care if it takes it longer to shut down as long as it is guaranteed to publish complete metrics, because we rely on those metrics for business insights and for alerts.

Reply:
Also, just for the cronjob we configured `step` duration to 5 seconds.

Reply:
>In our use case we don't care how long does it take to shut down the process as long as it publishes the metrics. We can hide this behavior behind an option I guess?

I understand that but I think it would be a fair assumption saying that most of the users do care since it is highly likely that by default this will result in blocking the app from gracefully shutting down for a while, then a violent `SIGKILL` (~`kill -9`) of the app. By default closing the registry should be registered in a shutdown hook so once the JVM wants to terminate, it will try to close the registry and wait... and wait... until the scheduler/orchestrator gets bored that the app is not shutting down and removes the app from memory violently.

I think an opt-in config option could help but 

>The processes are probably short-lived not because people want them to be short-lived, but because the work they do is performed fast.

I'm not sure I understand this, but my point with mentioning the short-lived processes was pointing out that this change will block them being really short-lived.
Imagine an app that is deployed into a "serverless"/FaaS (Function as a Service) environment where the orchestrator can shut down the apps (JVMs) frequently (when they are idle). In this scenario, blocking the app from shutting down can be sub-optimal.

>because we rely on those metrics for business insights and for alerts.

This might be a disconnect. Micrometer is trying not to disrupt the application while it is doing its job. This means that it might not be a good choice for business metrics/accounting/financing because it can drop data points (i.e.: in case of a network issue).